### PR TITLE
Add table oci_identity_tenancy. Closes #105

### LINKED ISSUE #105: Add table oci_identity_tenancy
**References**

This will help to query CIS > 3.1 Ensure audit log retention period is set to 365 days 

**Other reference link**
https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/managingtenancy.htm

image

### PR DESCRIPTION
**Pull request by visiit**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Test names: [ 'tests/oci_identity_tenancy' ]
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_identity_tenancy []

PRETEST: tests/oci_identity_tenancy

TEST: tests/oci_identity_tenancy
Running terraform
null_resource.retention_period: Creating...
null_resource.named_test_resource: Creating...
null_resource.retention_period: Provisioning with 'local-exec'...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci iam tenancy get --tenancy-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq > /private/var/folders/xc/kxfp1nw938dgs1ypvslzdsyc0000gr/T/tests/oci_identity_tenancy/terraform/test/identity_tenancy.json"]
null_resource.retention_period (local-exec): Executing: ["/bin/sh" "-c" "oci audit config get --compartment-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq > /private/var/folders/xc/kxfp1nw938dgs1ypvslzdsyc0000gr/T/tests/oci_identity_tenancy/terraform/test/retention_period.json"]
null_resource.named_test_resource: Still creating... [10s elapsed]
null_resource.retention_period: Still creating... [10s elapsed]
null_resource.named_test_resource (local-exec): WARNING: Permissions on /Users/vishalkumar/.oci/config are too open.
null_resource.retention_period (local-exec): WARNING: Permissions on /Users/vishalkumar/.oci/config are too open.
null_resource.retention_period (local-exec): To fix this please try executing the following command:
null_resource.named_test_resource (local-exec): To fix this please try executing the following command:
null_resource.named_test_resource (local-exec): oci setup repair-file-permissions --file /Users/vishalkumar/.oci/config
null_resource.retention_period (local-exec): oci setup repair-file-permissions --file /Users/vishalkumar/.oci/config
null_resource.retention_period (local-exec): Alternatively to hide this warning, you may set the environment variable, OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING:
null_resource.retention_period (local-exec): export OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING=True

null_resource.named_test_resource (local-exec): Alternatively to hide this warning, you may set the environment variable, OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING:
null_resource.named_test_resource (local-exec): export OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING=True

null_resource.retention_period: Creation complete after 15s [id=1253223344934529443]
data.local_file.periodInput: Refreshing state...
null_resource.named_test_resource: Creation complete after 15s [id=5543148077035589993]
data.local_file.input: Refreshing state...

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

description = lalit
resourceName = turbot
resource_id = ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq
retention_period_days = 365
tenancy_ocid = ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq

Running SQL query: test-hydrate-query.sql

[
  {
    "id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "name": "turbot",
    "retention_period_days": 365
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "description": "lalit",
    "id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "name": "turbot",
    "retention_period_days": 365
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "turbot"
  }
]
✔ PASSED

POSTTEST: tests/oci_identity_tenancy

TEARDOWN: tests/oci_identity_tenancy

SUMMARY:

1/1 passed.
Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "turbot"
  }
]
✔ PASSED

POSTTEST: tests/oci_identity_tenancy

TEARDOWN: tests/oci_identity_tenancy

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
- Basic info

select
  name,
  id,
  retention_period_days,
  description
from
  oci_identity_tenancy;

+--------+---------------------------------------------------------------------------------+-----------------------+-------------+
| name   | id                                                                              | retention_period_days | description |
+--------+---------------------------------------------------------------------------------+-----------------------+-------------+
| turbot | ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq | 365                   | lalit       |
+--------+---------------------------------------------------------------------------------+-----------------------+-------------+

- List tenancies where retention period is not set to 365 days

select
  name,
  id,
  retention_period_days,
  home_region_key
from
  oci_identity_tenancy
where
  retention_period_days <> 365;

+------+----+-----------------------+-----------------+
| name | id | retention_period_days | home_region_key |
+------+----+-----------------------+-----------------+
+------+----+-----------------------+-----------------+
```
</details>

### CODE REVIEW

File: oci-test/tests/oci_identity_tenancy/test-hydrate-query.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+select retention_period_days
```
Comment (khushboo9024) [inline oci-test/tests/oci_identity_tenancy/test-hydrate-query.sql:1]: Add some more columns like name, id also.

File: oci-test/tests/oci_identity_tenancy/variables.tf [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  depends_on = [null_resource.named_test_resource]
+  value      = lookup(lookup(jsondecode(data.local_file.input.content), "data", "data"), "name", "name")
+}
+
+output "description" {
+  depends_on = [null_resource.named_test_resource]
+  value      = lookup(lookup(jsondecode(data.local_file.input.content), "data", "data"), "description", "description")
+}
+
+output "retention_period_days" {
```
Comment (khushboo9024) [inline oci-test/tests/oci_identity_tenancy/variables.tf:69]: Is the local file generated by terraform getting deleted?

File: docs/tables/oci_identity_tenancy.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Table: oci_identity_tenancy
```
Comment (khushboo9024) [inline docs/tables/oci_identity_tenancy.md:1]: In issue table name is `oci_administration_tenancy_details` and table created is `oci_identity_tenancy` ?

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Please check the commnets

File: docs/tables/oci_identity_tenancy.md (hunk 3 above)
Comment (visiit) [reply docs/tables/oci_identity_tenancy.md:1]: @khushboo9024  In terraform data source, getting the like `oci_identity_tenancy`.
https://registry.terraform.io/providers/hashicorp/oci/latest/docs/data-sources/identity_tenancy

File: oci/table_oci_identity_tenancy.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+	}
+
+	request := audit.GetConfigurationRequest{
+		CompartmentId: &session.TenancyID,
+		RequestMetadata: oci_common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(),
+		},
+	}
+
+	response, err := session.AuditClient.GetConfiguration(ctx, request)
```
[the remark this answers is not in the document]
Comment (khushboo9024) [reply oci/table_oci_identity_tenancy.go:146]: we have removed `oci_audit_configuration` and keeping `oci_identity_tenancy` table .

**Review by khushboo9024 — APPROVED** [review]
LGTM

File: docs/tables/oci_identity_tenancy.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+select
+  name,
+  id,
+  retention_period_days,
+  description
+from
+  oci_identity_tenancy;
+```
+
+### List tenancies where retention period is set to 365 days
```
Comment (rajmohanty17) [inline docs/tables/oci_identity_tenancy.md:19]: We should show something as below

List tenancies where retention period is not set to 365 days
```sqlselect  name,  id,  retention_period_days,  home_region_keyfrom  oci_identity_tenancywhere  retention_period_days <> 365;```

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
Small change requested

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**